attributioncode/validator.go

```diff
@@ -13,10 +13,19 @@ import (
 )
 
 var validAttributionKeys = map[string]bool{
-	"source":   true,
-	"medium":   true,
-	"campaign": true,
-	"content":  true,
+	"source":            true,
+	"medium":            true,
+	"campaign":          true,
+	"content":           true,
+	"funnel_experiment": true,
+	"funnel_variation":  true,
```

Comment:
@mixedpuppy just making sure, on windows, the expected keys have no `funnel_`? https://hg.mozilla.org/releases/mozilla-release/file/2c8be14a89c92e56dbfc10ad32284e628a517a93/browser/components/attribution/AttributionCode.jsm#l66

Reply:
We should aim for consistency here. `funnel_` is not adding anything to these parameters. If Firefox expects `experiment` and `variation`, then we should plan on those being sent all the way from the start. Bedrock work hasn't begun, so we can get this in.

Reply:
IIRC funnel_ was requested.  On windows portions are stripped somewhere along the way to reduce data size I think, but they are not for osx (which doesn't work anyway, I noticed Mardak comment on some bug about using the quarantine data, that's what we tried).

Reply:
Looks like this was the last comment in the firefox bug:
https://bugzilla.mozilla.org/show_bug.cgi?id=1515172#c37

This says the url may contain "utm_" and "funnel_" that then end up getting stripped before stored as attribution.

Looking at the AttributionCode.jsm logic, it does seem to support "utm_" and "funnel_" and no-prefix for url-based attribution for mac while on windows, it only wants the stripped versions, e.g., "source" "medium" as before and the new "experiment"

Reply:
prefixed is supported of osx.  non-prefixed is for windows.  The stub installer is limited to 200 characters.

https://github.com/mozilla/bedrock/issues/7474#issuecomment-513291225

Reply:
> prefixed is supported of osx. non-prefixed is for windows. The stub installer is limited to 200 characters.
> 

Let's walk backwards:
* does telemetry require any particular param format?
* does the stub binary require any particular param format?
* can we simplify to avoid different behavior per platform?

Since we're working on the stubattribution service here, presumably we can plan to support those requirements. 

And since we're working on Bedrock in https://github.com/mozilla/bedrock/issues/7474, we can plan to support them there, too.

> [mozilla/bedrock#7474 (comment)](https://github.com/mozilla/bedrock/issues/7474#issuecomment-513291225)

Where, specifically, is the constraint? In https://bugzilla.mozilla.org/show_bug.cgi?id=1406005#c49, mhowell says the stub binary is limited to 1010. In this PR, we can see that it _was_ limited to 200, and is now set to 400, and I'm requesting 600-800. Are there any other hard constraints to be aware of?

Reply:
I've no opinions here, just attempting to relay what I can remember (hopefully relatively accurate) from when I looked at this last...

I didn't run into anything with specific limits/formats in the telemetry data.

To my knowledge the windows stub binary wouldn't require any specific format.  The download service that builds the stub might, and IIRC that is a) where the prefix is stripped, b) has a hard coded list of what keys are allowed into the stub.

I'd verify in current code what the current limit is.  Seems like 1000 as mentioned should be enough.

osx attributions (if a workaround is ever figured out) happen via the quarantine database, so the param keys are unchanged from whatever the referrer url was.  That is why the service tries to support both with/without prefix.

Reply:
It looks like the time this logic here (stubattribution validator.go) is used, there's a combined `attribution_code` that is parsed and handles keys like "source" and pretty sure there's no knowledge of "utm" anywhere in this repository.

From my searching around, stripping of "utm_" happens on bedrock backend as part of the hmac signature generation process which currently accepts `{utm_source, utm_medium, …}` json and returns `{attribution_code, attribution_sig}` json where the "attribution_code" has the concatenated prefix-stripped keys.

https://github.com/mozilla/bedrock/blob/67a2debdfcbd4a72cc3b5b0998a136aea7964704/bedrock/firefox/views.py#L95-L96

So yes it looks like validator.go should use "experiment" and "variation" without "funnel_" just like it doesn't currently expect "utm_*". And these prefix-less attribution will work with existing windows firefox 71.

(And attribution doesn't work quite right on osx anyway, and doesn't get called from bedrock anyway, so we don't really need to worry about compatibility yet. https://bugzilla.mozilla.org/show_bug.cgi?id=1525076 )